build failing randomly

I've been using this jenkins plugin for over 6 months to push to my elastic beanstalk applications and it has been working flawlessly until recently.

Here is the error I have been getting.

-------------- 

 ERROR: Build step failed with exception
java.lang.RuntimeException: java.lang.NullPointerException
	at org.jenkinsci.plugins.awsbeanstalkpublisher.extensions.AWSEBSetup.perform(AWSEBSetup.java:25)
	at org.jenkinsci.plugins.awsbeanstalkpublisher.AWSEBBuilder.perform(AWSEBBuilder.java:65)
	at hudson.tasks.BuildStepMonitor$1.perform(BuildStepMonitor.java:20)
	at hudson.model.AbstractBuild$AbstractBuildExecution.perform(AbstractBuild.java:779)
	at hudson.model.Build$BuildExecution.build(Build.java:205)
	at hudson.model.Build$BuildExecution.doRun(Build.java:162)
	at hudson.model.AbstractBuild$AbstractBuildExecution.run(AbstractBuild.java:534)
	at hudson.model.Run.execute(Run.java:1728)
	at hudson.model.FreeStyleBuild.run(FreeStyleBuild.java:43)
	at hudson.model.ResourceController.execute(ResourceController.java:98)
	at hudson.model.Executor.run(Executor.java:404)
Caused by: java.lang.NullPointerException
	at org.jenkinsci.plugins.awsbeanstalkpublisher.AWSEBS3Uploader.uploadArchive(AWSEBS3Uploader.java:77)
	at org.jenkinsci.plugins.awsbeanstalkpublisher.AWSEBEnvironmentUpdater.perform(AWSEBEnvironmentUpdater.java:66)
	at org.jenkinsci.plugins.awsbeanstalkpublisher.extensions.AWSEBElasticBeanstalkSetup.perform(AWSEBElasticBeanstalkSetup.java:174)
	at org.jenkinsci.plugins.awsbeanstalkpublisher.extensions.AWSEBSetup.perform(AWSEBSetup.java:21)
	... 10 more
Build step 'Deploy into AWS Elastic Beanstalk' marked build as failure

-------------- 

I first got the following error about a month ago and I could not deploy to my elastic beanstalk application. But a few hours later, when I deployed again without making changes, it was able to be deployed successfully. 

About 3 weeks later, I got this error again but for another job. Like before, it started working fine a few hours later. 

Because it started working on again without any changes, I thought it was maybe an issue on the AWS S3 end so I contacted the aws support but unfortunately they could not identify any problems...

Today, another job failed while deploying to elastic beanstalk. This time, I decided to debug deeper to see if I can find the root cause of this. I ended up cloning the failed job and then tried to deploy with the cloned job but failed due to the version name conflict. After changing the name of the version, it was able to deploy successfully. 

I went back to my original job and tried to deploy without making any changes, but like before, it failed again. I then changed the version name and re-deployed and this time, it was successful. 

Do you know how and why that fixed this issue?  I just thought i'd create an issue here just in case anyone has come across this issue before.

--- a/src/main/java/org/jenkinsci/plugins/awsbeanstalkpublisher/extensions/AWSEBSetup.java
+++ b/src/main/java/org/jenkinsci/plugins/awsbeanstalkpublisher/extensions/AWSEBSetup.java
@@ -18,7 +18,9 @@ public abstract class AWSEBSetup extends AbstractDescribableImpl<AWSEBSetup> {
         boolean status = true;
         try {
             for (AWSEBSetup eb : extensions) {
-                status &= eb.perform(build, launcher, listener);
+                if (eb != null) {
+                    status &= eb.perform(build, launcher, listener);
+                }
             }
             return status;
         } catch (Exception exc) {

